pkg/executor/local.go

```diff
@@ -255,3 +277,64 @@ func (l *Local) copyFile(src, dst string) error {
 
 	return nil
 }
+
+func (l *Local) deletePath(ctx context.Context, src string, excl []string) error {
+	info, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+
+	if !info.IsDir() || len(excl) == 0 {
+		return os.RemoveAll(src)
+	}
+
+	hasExclusion := false
+	err = filepath.Walk(src, func(filePath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+
+		relPath, err := filepath.Rel(src, filePath)
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() && isExcludedSubPath(relPath, excl) {
+			return nil
+		}
+
+		if isExcluded(relPath, excl) {
+			hasExclusion = true
+			if info.IsDir() {
+				return filepath.SkipDir
+			}
+
+			return nil
+		}
+
+		if !info.IsDir() {
+			return os.Remove(filePath)
+		}
+
+		err = os.RemoveAll(filePath)
+		if err != nil {
+			return err
+		}
+
+		return filepath.SkipDir
+	})
+
+	if err != nil {
+		return err
+	}
+
+	if !hasExclusion {
+		return os.RemoveAll(src)
```

Comment:
this part is confusing me a little bit. the walk above removing things too, but probably something left? If this is the case a comment explaining this part will be nice

Reply:
The idea is to remove the top-level directory if there are no actual excursions. For instance, when an `exclude` list is provided but there are no files that matched the items in the list. In such a scenario `walk` is going to bypass the top-level dir `./`.
Does it make sense for you?